### PR TITLE
Remove escape function.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/ast/ASTPrinter.cpp [hunk 1]
```diff
@@ -579,8 +582,11 @@ void ASTPrinter::printSourcePart(ASTNode const& _node)
 	if (!m_source.empty())
 	{
 		SourceLocation const& location(_node.location());
-		*m_ostream << indentation() << "   Source: "
-				   << escaped(m_source.substr(location.start, location.end - location.start), false) << endl;
+		*m_ostream <<
+			indentation() <<
+			"   Source: " <<
+			jsonCompactPrint(m_source.substr(location.start, location.end - location.start)) <<
```
Comment (axic) [inline libsolidity/ast/ASTPrinter.cpp]: See https://github.com/ethereum/solidity/pull/2801#discussion_r135225060

**Comment by chriseth** [comment]
@axic can this be merged?

File: libsolidity/ast/ASTPrinter.cpp (hunk 1 above)
Comment (axic) [reply libsolidity/ast/ASTPrinter.cpp]:
```
JSONCPP_STRING write(const Value& root) JSONCPP_OVERRIDE;
```

So I guess it creates a copy of the string?

I'd prefer if we could just do it with a simple regex. Is that possible?

Comment (chriseth) [reply libsolidity/ast/ASTPrinter.cpp]: What is the advantage? I would prefer not to create our own hand-written code if we already have code that does exactly the same thing in a library.

**Comment by axic** [comment]
Confirmed the output is the same:
```
Source: "contract Test {\nfunction Test(){\nmsg.sender.call.value(0);\nmsg.sender.call.value(0)();\n}\n}"
```

Though it would be more appropriate to use `toString` and not compact writer, because that supposed to be for proper JSON (which cannot be a string). And it also seems to make a copy of the string.

**Review by axic — APPROVED** [review]

**Comment by axic** [comment]
Hm, actually there is such a function, `valueToQuotedString` and it does no copy - it essentially is the same as our escaping.

**Comment by axic** [comment]
@chriseth revert the last commit if you don't like it and merge it that way

**Comment by chriseth** [comment]
Ah ok, I saw valueToQuotedString, but I thought it was not part of the API.

**Pull request merged**